soci/fbs/ztoc.fbs

```diff
@@ -0,0 +1,48 @@
+namespace ztoc;
+
+table Xattr {
+	key : string;
+	value : string;
+}
+
+table FileMetadata {
+	name : string;
+	type : string;
+	uncompressedOffset : long;
+	uncompressedSize : long;
+	spanStart : int;
+	spanEnd : int;
+	firstSpanHasBits : bool;	// Flag for if there is partial uncompressed data that is stored in the previous byte
+	linkname : string;		// Target name of link (valid for TypeLink or TypeSymlink)
+	mode : long;			// Permission and mode bits
+	uid : int;			// User ID of owner
+	gid : int;			// Group ID of owner
+	uname : string;			// User name of owner
+	gname : string;			// Group name of owner
+
+	modTime  : string;		// Modification time (is formatted in RFC 3339 format,
+					// with sub-second precision added if present: "2006-01-02T15:04:05.999999999Z07:00")
+	devmajor : long;		// Major device number (valid for TypeChar or TypeBlock)
+	devminor : long;		// Minor device number (valid for TypeChar or TypeBlock)
+
+	xattrs : [Xattr];
+}
+
+table ZtocInfo {
```

Comment:
As a bit of future-proofing, let's add an enum for compression algorithm. We only support gzip today. But Zstandard is coming.

Reply:
Let's rename this so that it is clear that it relates to compression. We've gotten ourselves in trouble by using a cutsey name like "zTOC", because it's left us without a good name to refer specifically to the compression half of the format.

Let's fix that once and for all. Either double down on the cutsiness, and call this `ZInfo` or something like that; call it `CompressionInfo`. I prefer the latter. Either way, let's only ever use the four letters z-t-o-c when we are referring to the entire thing; both halves. And let's only ever use the three letters t-o-c when we are referring to the tarball table of contents.

Reply:
No way to use `struct` there, since it allows for only scalar values and other structs.  

> As a bit of future-proofing, let's add an enum for compression algorithm. We only support gzip today. But Zstandard is coming.

Done.

> Let's fix that once and for all. Either double down on the cutsiness, and call this `ZInfo` or something like that; call it `CompressionInfo`. I prefer the latter. Either way, let's only ever use the four letters z-t-o-c when we are referring to the entire thing; both halves. And let's only ever use the three letters t-o-c when we are referring to the tarball table of contents.

Done.